api/custom_auth/urls.py

```diff
@@ -26,7 +26,11 @@
         CustomAuthTokenLoginWithMFACode.as_view(),
         name="mfa-authtoken-login-code",
     ),
-    path("logout/", TokenDestroyView.as_view(), name="authtoken-logout"),
+    path(
+        "logout/",
+        JWTSlidingBlacklistView.as_view(),
+        name="authtoken-logout",
+    ),
```

Comment:
So, I guess what you're saying in other conversations is that `TokenDestroyView` does nothing, is that right? I guess I'm concerned about a couple of things here: 

 1. Are we removing functionality here for regular tokens? I assume not, but just checking. 
 2. Shouldn't we just have a `Logout` view class where we can internalise the conditional logic depending on the setting?

Reply:
`TokenDestroyView`, in fact, does remove the token. However, the endpoint has never been used by the frontend; to log the user out, the application simply removed the `t` cookie.

Now that I expect the endpoint to be actually used, I ultimately decided to remove function related to classic tokens and only implement the JWT logout.

I have renamed the view and added a docstring to hopefully be more clear.

Reply:
Maybe the FE never used this functionality, but that's not to say that it's never been used by someone integrating with the API. It's obviously very unlikely, but not totally out of the question since we only provided API access via these user tokens for a long time. 

I don't see the harm in either: 

 1. Reverting to what you had initially and subclassing `TokenDestroyView` and calling `super()`
 2. Having 2 separate views that we add into urlpatterns depending on the existence of the environment variable.

Reply:
>They will still have the DELETE /api/v1/auth/token/ endpoint that serves this exact function, and arguably has better semantics as we do not delete classic tokens on logout.

But it's technically a breaking API change, right? We're removing functionality from an existing endpoint, that existed, and was documented for people to use.

Reply:
We can't just remove it though. I don't mind adding a warning that it'll be deprecated in the next version (and let's add it to the V2 API issue).

Reply:
The problem is that it's being called in FE now. We need to either add conditional logic to FE or explain to users that their tokens get invalidated if they login explicitly. The latter is fine by me as using an admin API key is the clear alternative.

Reply:
> their tokens get invalidated if they login explicitly

I assume here you meant to say 'logout explicitly' ? 

It doesn't seem like it'd be hard to add conditional logic to the frontend to only call the logout endpoint if the jwt cookie exists, and seems to be the right thing to do to maintain compatibility, but I can't speak for @kyle-ssg here. 

Alternatively, we do what you suggested, and just show a warning when the user hits the logout button to say that any direct API integrations they have using the user token will stop functioning.

Another alternative (that doesn't involve any more FE work) is still to conditionally change the view class used by the `logout` route depending on the existence of environment variable, but we may then need to think about whether we can actually run both in parallel.